### PR TITLE
Document the manifest models

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
Our manifest models lacked documentation so I added a new page that explains what each attribute is for.

I'd appreciate a proof reading 😬

### CODE REVIEW

File: docs/usage/manifest.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+- **Settings (optional):** Read more about [settings](#settings)
+
+> **Note:** All the relative paths in the project manifest are relative to the folder that contains the manifest file.
+
+### Target
+
+Each target in the list of project targets can be initialized with the following attributes:
+
+- **Name:** The name of the target. The Xcode project target and the derivated product take the same name.
+- **Platform:** The platform the target product is build for. The following products are supported:
```
Comment (kwridan) [inline docs/usage/manifest.md]: > is built for

File: docs/usage/manifest.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  - **.app:** An application
+  - **.staticLibrary:** A static library.
```
Comment (kwridan) [inline docs/usage/manifest.md]: do we want to call out `.staticFramework` ?

if we do, I'd add a note:

- **.staticFramework:** A static framework. This is a regular framework product however is configured to be staticky linked.

File: docs/usage/manifest.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+  - **Public:** Relative path to the folder that contains the public headers.
+  - **Private:** Relative path to the folder that contains the private headers.
+  - **Project:** Relative path to the folder that contains the project headers.
+
+- **Entitlements (optional):** Relative path to the entitlements file.
+- **Actions (optional):** Target actions allow defining extra script build phases. It's an array of `TargetAction` objects that can be initialized with any of the following static functions:
+
+  - **pre(tool:):** Runs the binary with the given name before the default target build phases.
+  - **pre(path:):** Runs the binary at the given path before the default target build phases.
```
Comment (kwridan) [inline docs/usage/manifest.md]: if I'm not mistaken the use case is:
`.pre(tool: "/bin/sh", path: "my_script.sh")`  (default is `/bin/sh`)

Is it worth for this one including examples to clarify what is meant by binary?

`.pre(...)`: Runs before the target's default build phase
`.post(...)`: Runs before the target's default build phase

Parameters:
- **tool (optional)**: The binary to use to execute the build phase.
- **path (optional)**: Path to the script to execute.
- **arguments**: Path to the script to execute.
- **name**: The name of the build phase

Examples:

`.pre(path: "my_custom_script.sh", arguments: ["hello"], name: "My Custom Script Phase")`
...

Comment (pepibumur) [reply docs/usage/manifest.md]: Good idea 👍

**Pull request merged**